### PR TITLE
bh - Adding backend functionality for name 15 char limit in personal schedule

### PR DESCRIPTION
**Pull request by bhuang01**
In this PR, we add backend functionality to restrict the name parameter when creating a new personal schedule. Returns a a 404 error code along with a message. This closes issue #11 

image
image

### CODE REVIEW

**Review by andrewpengucsb — APPROVED** [review]
LGTM, merge after #10

File: src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSchedulesController.java [hunk 1]
```diff
@@ -88,6 +88,11 @@ public PersonalSchedule postSchedule(
         CurrentUser currentUser = getCurrentUser();
         log.info("currentUser={}", currentUser);
 
+        // Checks length of name parameter (length should be 15 chars or less, nonzero)
+        if (name.length() > 15) {
```
Comment (eduglt) [inline src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSchedulesController.java:92]: When testing this code, we can pass the mutation tests that change the boundary value by adding two tests: one that should accept with `name.length() == 15` (the maximum allowed length), and one that should reject because `name.length() == 16` (the minimum disallowed length).

**Review by eduglt — DISMISSED** [review]
LGTM, but do we need tests?

File: src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSchedulesController.java (hunk 1 above)
Comment (pconrad) [reply src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSchedulesController.java:92]: Yes... you do need tests... :-)

**Comment by pconrad** [comment]
Still missing a test for: 

image

**Review by eduglt — DISMISSED** [review]
LGTM

**Review by eduglt — DISMISSED** [review]
LGTM, successful merge with main branch.

**Review by pconrad — DISMISSED** [review]
LGTM... will want to test on qa before merging

**Comment by eduglt** [comment]
Tested Bryan's on Heroku one more time, LGTM on there too!

File: src/main/java/edu/ucsb/cs156/courses/repositories/PersonalScheduleRepository.java [hunk 2]
```diff
@@ -12,5 +12,4 @@
 public interface PersonalScheduleRepository extends CrudRepository<PersonalSchedule, Long> {
   Optional<PersonalSchedule> findByIdAndUser(long id, User user);
   Iterable<PersonalSchedule> findAllByUserId(Long user_id);
-  Optional<PersonalSchedule> findByUserAndNameAndQuarter(User user, String name, String quarter);
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/courses/repositories/PersonalScheduleRepository.java:15]: Why are you removing this?     Removing code needs a justification.  Are you absolutely sure that it's not being used anywhere else in the code base?

**Review by pconrad — CHANGES_REQUESTED** [review]
I think I can see why there are test failures on CI/CD ... I think you may have stomped on some other folks' necessary code.

**Review by eduglt — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM... will test on qa
